source/reflect.h2

```diff
@@ -490,9 +551,15 @@ type_declaration: @copyable type =
     add_member: (inout this, source: std::string_view)
     = {
         decl := parse_statement(source);
-        if !decl || !n*.add_type_member(decl) {
-            error( std::string("error attempting to add member:\n") + source );
+        if !(decl as bool) {
+            error("the provided source string is not a valid statement");
```

Comment:
```suggestion
            error("the provided source string is not a valid statement:\n(source)$" );
```
This one used to include the source fragment.
This is where I'm getting the error, but I can't as easily see what's wrong:
```output
algorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: invalid else branch body (at '{')
algorithms.test.cpp2: error: ill-formed initializer (at '{')
algorithms.test.cpp2(37,1): error: while applying @array - the provided source string is not a valid statement
```
With the suggestion above (`add_initializer` already does it right):
```output
algorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: unexpected text '
lgorithms.test.cpp2: error: invalid else branch body (at '{')
algorithms.test.cpp2: error: ill-formed initializer (at '{')
algorithms.test.cpp2(37,1): error: while applying @array - the provided source string is not a valid statement:
operator(): <I, S> (this, copy first: I, copy last: S)
  -> std::iter_difference_t<I> pre<Bounds>(0 <= n) requires std::sentinel_for<S, I> && std::totally_ordered_with<decltype((n)), std::iter_difference_t<I>> = {
  if constexpr std::sized_sentinel_for<S, I> {
  if constexpr std::same_as<S, I> {
    last = std::ranges::next(first, n, last);
  } else {
    return operator()(first, std::ranges::next(first, n, last));
  }
}
take := n;
if constexpr std::sized_sentinel_for<S, I> {
  return last - first;
} else {
    return take - std::ranges::advance(first, take, last);error: expected a single string_ = first;
    
  }
}
```
I can immediately see what's wrong in color:
![1702703535](https://github.com/hsutter/cppfront/assets/21071787/b347b947-6874-4165-a3d5-9c54ec3db112)

Reply:
By the way, to print in colors, I use <https://stackoverflow.com/questions/5947742/how-to-change-the-output-color-of-echo-in-linux>
It's very useful, since I'm programmatically generating the source code.

```Cpp2
warn: (forward s) = _ = std::cerr << "\033[1;35mwarning:\033[0m (s)$\n";

error: (forward s) -> std::string = "\033[1;31m(s)$\033[0m";
```

```Cpp2
single_string: type == split_strings<"\033[1;31merror: expected a single string\033[0m">;
```

![1702704113](https://github.com/hsutter/cppfront/assets/21071787/90972c4b-db49-4664-9223-24ca7a310408)

Reply:
> By the way, to print in colors, I use https://stackoverflow.com/questions/5947742/how-to-change-the-output-color-of-echo-in-linux
> It's very useful, since I'm programmatically generating the source code.

Thanks. Are you suggesting that cppfront have something like this, possibly under a new `-fcolor-diagnostics` or `-fdiagnostics-color` switch?

Reply:
Sorry, I missed your reply.
No, that wasn't my intention.
But I might come back later for that.